### PR TITLE
Adding ScrollController support for Stepper widget

### LINKED ISSUE #61207: Vertical Stepper should have scrollController field
## Use case

For other components that are intended to wrap `Scrollable` widgets, like [DraggableScrollableSheet](https://api.flutter.dev/flutter/widgets/DraggableScrollableSheet-class.html), it's necessary to specify the `scrollController`.

I'm currently using this to specify a `scrollController`, however, it adds an extra, otherwise unnecessary `SingleChildScrollView`:

```dart
             SingleChildScrollView(
                controller: scrollController,
                child: Stepper(
                  physics: ClampingScrollPhysics(),
                  ...
                ),
              ),
```

## Proposal

Ideally, it'd be nice to simply use this instead:

```dart
             Stepper(
                scrollController: scrollController,
                ...
             ),
```

### PR DESCRIPTION
**Pull request by SwaroopSambhayya**
This PR is to add **controller** property to the **Stepper** widget, so that user has the flexibility to control the scroll offset for various purposes(especially in case of scroll animations)!

Fixes #61207 

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by HansMuller — COMMENTED** [review]
This looks like a reasonable change. Will need some fixups for the analyzer and test failures.

**Comment by SwaroopSambhayya** [comment]
@HansMuller as I'm new can you point me to any docs to the fixups for analyser and test failures? seems the issue is not related to my change

File: packages/flutter/test/material/stepper_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/stepper_test.dart:975]:
```suggestion
  testWidgets('ScrollController is passed to the stepper listview', (WidgetTester tester) async {
```

File: packages/flutter/test/material/stepper_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/stepper_test.dart:976]: Analyzer is failing here
```suggestion
      final ScrollController controller = ScrollController();
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/stepper_test.dart
Comment (SwaroopSambhayya) [reply packages/flutter/test/material/stepper_test.dart:976]: Okay got it! thanks

File: packages/flutter/test/material/stepper_test.dart
Comment (SwaroopSambhayya) [reply packages/flutter/test/material/stepper_test.dart]: Added the change

File: packages/flutter/test/material/stepper_test.dart
Comment (SwaroopSambhayya) [reply packages/flutter/test/material/stepper_test.dart]: Added the change

File: packages/flutter/test/material/stepper_test.dart
Comment (SwaroopSambhayya) [reply packages/flutter/test/material/stepper_test.dart]: Added change

File: packages/flutter/lib/src/material/stepper.dart
Comment (TahaTesser) [inline packages/flutter/lib/src/material/stepper.dart:236]:
```suggestion
  /// An object that can be used to control the position to which this scroll
  /// view is scrolled.
  ///
  /// To control the initial scroll offset of the scroll view, provide a
  /// [controller] with its [ScrollController.initialScrollOffset] property set.
  final ScrollController? controller;
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]
Update doc to be consistent.

**Review by TahaTesser — APPROVED** [review]
LGTM!

This will need a second review from @HansMuller before landing.

**Pull request merged**
